### ISSUE TITLE
[consistent-type-imports] Regression 4.14.0/1 error when import used as key in type object

### ISSUE THREAD

**Issue by lukeapage** [typescript-eslint/typescript-eslint#2989]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/consistent-type-imports": "error"
  }
}
```

```TS
import type * as constants from './constants';

export type Y = {
    [constants.X]: ReadonlyArray<string>;
};
```

**Expected Result**

No error as in 4.13.0

**Actual Result**

error:
```
 1:1  error  Type import "constants" is used by decorator metadata  @typescript-eslint/consistent-type-imports
```
**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `4.14.1` |
| `@typescript-eslint/parser`        | `4.14.1` |
| `TypeScript`                       | `4.1.3` |
| `ESLint`                           | `7.19.0` |
| `node`                             | `12.16.0` |
